Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py

```diff
@@ -228,13 +233,17 @@ def calculate_fetch_start_time(last_fetch_time: datetime | str | None, fetch_sta
             last_fetch_datetime = last_fetch_time
         if not last_fetch_datetime:
             raise DemistoException(f'last_fetch_time is not a valid date: {last_fetch_time}')
-        return last_fetch_datetime
+        if last_fetch_datetime > before_time:
+            last_fetch_datetime = before_time - timedelta(minutes=1)
+        return last_fetch_datetime, before_time
     # case 2
     elif fetch_start_time:
-        return fetch_start_time
+        if fetch_start_time > before_time:
+            last_fetch_datetime = before_time - timedelta(minutes=1)
+        return fetch_start_time, before_time
     # case 3
     else:
-        return datetime.now() - timedelta(minutes=1)
+        return before_time - timedelta(minutes=1), before_time
```

Comment:
just return the before_time. do `- timedelta(minutes=1) ` when using this instead

Reply:
I'll make something less code duplicating.